File: android/app/src/main/java/info/bagen/dwebbrowser/microService/sys/nativeui/NativeUiNMM.kt
package info.bagen.dwebbrowser.microService.sys.nativeui

import info.bagen.dwebbrowser.microService.core.BootstrapContext
import info.bagen.dwebbrowser.microService.core.NativeMicroModule
import info.bagen.dwebbrowser.microService.sys.nativeui.navigationBar.NavigationBarNMM
import info.bagen.dwebbrowser.microService.sys.nativeui.safeArea.SafeAreaNMM
import info.bagen.dwebbrowser.microService.sys.nativeui.splashScreen.SplashScreenNMM
import info.bagen.dwebbrowser.microService.sys.nativeui.statusBar.StatusBarNMM
import info.bagen.dwebbrowser.microService.sys.nativeui.virtualKeyboard.VirtualKeyboardNMM

class NativeUiNMM : NativeMicroModule("nativeui.browser.dweb") {
    private val navigationBarNMM = NavigationBarNMM()
    private val statusBarNMM = StatusBarNMM()
    private val safeAreaNMM = SafeAreaNMM()
    private val virtualKeyboardNMM = VirtualKeyboardNMM()
    private  val splashScreenNMM = SplashScreenNMM()

    override suspend fun _bootstrap(bootstrapContext: BootstrapContext) {
        bootstrapContext.dns.install(navigationBarNMM)
        bootstrapContext.dns.install(statusBarNMM)
        bootstrapContext.dns.install(safeAreaNMM)
        bootstrapContext.dns.install(virtualKeyboardNMM)
        bootstrapContext.dns.install(splashScreenNMM)
    }

    override suspend fun _shutdown() {
    }
}

File: next/dweb-browser/src/Platforms/iOS/MicroModule/Browser/NativeUI/NativeUiNMM.cs
using DwebBrowser.MicroService.Browser.NativeUI.NavigationBar;
using DwebBrowser.MicroService.Browser.NativeUI.SafeArea;
using DwebBrowser.MicroService.Browser.NativeUI.StatusBar;
using DwebBrowser.MicroService.Browser.NativeUI.VirtualKeyboard;

namespace DwebBrowser.MicroService.Browser.NativeUI;

public class NativeUiNMM : NativeMicroModule
{
    public NativeUiNMM() : base("nativeui.browser.dweb", "nativeui")
    {
    }

    private readonly StatusBarNMM _statusBarNMM = new();
    private readonly NavigationBarNMM _navigationBarNMM = new();
    private readonly VirtualKeyboardNMM _virtualKeyboardNMM = new();
    private readonly SafeAreaNMM _safeAreaNMM = new();

    protected override async Task _bootstrapAsync(IBootstrapContext bootstrapContext)
    {
        bootstrapContext.Dns.Install(_statusBarNMM);
        bootstrapContext.Dns.Install(_navigationBarNMM);
        bootstrapContext.Dns.Install(_virtualKeyboardNMM);
        bootstrapContext.Dns.Install(_safeAreaNMM);
    }
}

